collector/slave_hosts.go

```diff
@@ -63,7 +63,17 @@ func (ScrapeSlaveHosts) Version() float64 {
 
 // Scrape collects data from database connection and sends it over channel as prometheus metric.
 func (ScrapeSlaveHosts) Scrape(ctx context.Context, db *sql.DB, ch chan<- prometheus.Metric, logger log.Logger) error {
-	slaveHostsRows, err := db.QueryContext(ctx, slaveHostsQuery)
+	var (
+		slaveHostsRows *sql.Rows
+		err            error
+	)
+	// Try the both syntax for MySQL 8.0 and MySQL 8.4
+	for _, query := range slaveHostsQueries {
```

Comment:
I'm not sure I like this for loop/break pattern. I would prefer to do a simple linear `if` drop through.